vmArgs in launch.json does not accept an array of strings

When configuring a java debugger target, in launch.json, it seems that vmArgs does not accept an array of strings. Note that I've configured vmArgs in the linux section as well.

##### Environment
- Operating System: Linux
- JDK version: 13
- Visual Studio Code version: latest as of 2020-03-16
- Java extension version: I've installed the java pack.

##### Backtrace:
```[Error - 2:14:41 PM] Mar 16, 2020, 2:14:41 PM Dispatch debug protocol error: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected STRING but was BEGIN_ARRAY at path $.vmArgs
java.lang.IllegalStateException: Expected STRING but was BEGIN_ARRAY at path $.vmArgs
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected STRING but was BEGIN_ARRAY at path $.vmArgs
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:224)
	at com.google.gson.Gson.fromJson(Gson.java:888)
	at com.google.gson.Gson.fromJson(Gson.java:953)
	at com.google.gson.Gson.fromJson(Gson.java:926)
	at com.microsoft.java.debug.core.protocol.JsonUtils.fromJson(JsonUtils.java:34)
	at com.microsoft.java.debug.core.adapter.DebugAdapter.dispatchRequest(DebugAdapter.java:76)
	at com.microsoft.java.debug.core.adapter.ProtocolServer.dispatchRequest(ProtocolServer.java:118)
	at com.microsoft.java.debug.core.protocol.AbstractProtocolServer.lambda$new$0(AbstractProtocolServer.java:78)
	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:60)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:200)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: java.lang.IllegalStateException: Expected STRING but was BEGIN_ARRAY at path $.vmArgs
	at com.google.gson.internal.bind.JsonTreeReader.nextString(JsonTreeReader.java:180)
	at com.google.gson.internal.bind.TypeAdapters$16.read(TypeAdapters.java:401)
	at com.google.gson.internal.bind.TypeAdapters$16.read(TypeAdapters.java:389)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220)
	... 17 more

```

--- a/src/configurationProvider.ts
+++ b/src/configurationProvider.ts
@@ -16,6 +16,13 @@ import { logger, Type } from "./logger";
 import * as utility from "./utility";
 import { VariableResolver } from "./variableResolver";
 
+const platformNameMappings = {
+    "win32": "windows",
+    "linux": "linux",
+    "darwin": "osx"
+};
+const platformName = platformNameMappings[process.platform];
+
 export class JavaDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
     private isUserSettingsDirty: boolean = true;
     private debugHistory: MostRecentlyUsedHistory = new MostRecentlyUsedHistory();
@@ -46,6 +53,9 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         vscode.ProviderResult<vscode.DebugConfiguration> {
         const resolveDebugConfigurationHandler = instrumentOperation("resolveDebugConfiguration", (operationId: string) => {
             try {
+                // See https://github.com/microsoft/vscode-java-debug/issues/778
+                // Merge the platform specific properties to the global config to simplify the subsequent resolving logic.
+                this.mergePlatformProperties(folder, config);
                 this.resolveVariables(folder, config);
                 return this.heuristicallyResolveDebugConfiguration(folder, config);
             } catch (ex) {
@@ -91,6 +101,19 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 }
             });
         });
+    }
+
+    private mergePlatformProperties(folder: vscode.WorkspaceFolder, config: vscode.DebugConfiguration) {
+        if (config && platformName && config[platformName]) {
+            try {
+                for (const key of Object.keys(config[platformName])) {
+                    config[key] = config[platformName][key];
+                }
+                config[platformName] = undefined;
+            } catch {
+                // do nothing
+            }
+        }
     }
 
     private resolveVariables(folder: vscode.WorkspaceFolder, config: vscode.DebugConfiguration): void {

--- a/src/configurationProvider.ts
+++ b/src/configurationProvider.ts
@@ -17,9 +17,9 @@ import * as utility from "./utility";
 import { VariableResolver } from "./variableResolver";
 
 const platformNameMappings = {
-    "win32": "windows",
-    "linux": "linux",
-    "darwin": "osx"
+    win32: "windows",
+    linux: "linux",
+    darwin: "osx",
 };
 const platformName = platformNameMappings[process.platform];
 

